From d3045184699d6220e5c03c2d6b1dfc449c3b873e Mon Sep 17 00:00:00 2001
From: Dan Carpenter <error27@gmail.com>
Date: Tue, 14 Jul 2026 18:50:18 +0300
Subject: [PATCH 07733/15333] clk: document that clk_get_parent() returns NULL

The documentation in the clk.h file says that clk_get_parent() returns
error pointers but it doesn't.  It's also not consistent with the
comments next to the clk_get_parent() implementation which say that
it returns NULL when the clk is NULL.  Update the comments so they
are consistent and accurate and say that it returns NULL.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Reviewed-by: Brian Masney <bmasney@redhat.com>
Signed-off-by: Brian Masney <bmasney@redhat.com>
---
 include/linux/clk.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/clk.h b/include/linux/clk.h
index 998ba3f261da..993fd6e916c7 100644
--- a/include/linux/clk.h
+++ b/include/linux/clk.h
@@ -951,8 +951,8 @@ int clk_set_parent(struct clk *clk, struct clk *parent);
  * clk_get_parent - get the parent clock source for this clock
  * @clk: clock source
  *
- * Returns struct clk corresponding to parent clock source, or
- * valid IS_ERR() condition containing errno.
+ * Returns struct clk corresponding to parent clock source, or NULL
+ * if clk is NULL.
  */
 struct clk *clk_get_parent(struct clk *clk);
 
-- 
2.55.0

